### ISSUE TITLE
Raise diagnostic for out-of-scope `ParamSpec` components

### ISSUE THREAD

**Issue by dhruvmanila** [astral-sh/ty#3742]
Ref https://github.com/astral-sh/ty/issues/1861

The following example should raise a diagnostic

```py
from typing import ParamSpec, Callable, Concatenate

P = ParamSpec("P")

# ty: no diagnostic
def bad(*args: P.args, **kwargs: P.kwargs) -> None: ...
```

[Playground](https://play.ty.dev/e31346f2-f958-4225-9dbd-44b140863fdd)

**Comment by charliermarsh** [comment]
We now reject this (via https://github.com/astral-sh/ruff/pull/27378).
